### ISSUE TITLE
Glances on a French Windows 8.1 workstation

### ISSUE THREAD

**Issue by mohierf** [nicolargo/glances#715]
I started some tests with Glances on several Linux distros and I now try to run on some Windows workstations ...

 One workstation is a French Windows 8.1, and I installed Python 2.7, Glances dependencies referenced in the readme Github page, and then `pip install Glances`. No problems for all these steps ...

 Then I start Glances: `glances` and get a Python exception: 
```
Traceback (most recent call last):
  File "c:\Python27\Scripts\glances-script.py", line 9, in <module>
    load_entry_point('Glances==2.5.1', 'console_scripts', 'glances')()
  File "C:\Python27\lib\site-packages\glances\__init__.py", line 125, in main
    standalone.serve_forever()
  File "C:\Python27\lib\site-packages\glances\core\glances_standalone.py", line
108, in serve_forever
    return self.__serve_forever()
  File "C:\Python27\lib\site-packages\glances\core\glances_standalone.py", line
93, in __serve_forever
    self.screen.update(self.stats)
  File "C:\Python27\lib\site-packages\glances\outputs\glances_curses.py", line 9
14, in update
    self.flush(stats, cs_status=cs_status)
  File "C:\Python27\lib\site-packages\glances\outputs\glances_curses.py", line 8
92, in flush
    self.display(stats, cs_status=cs_status)
  File "C:\Python27\lib\site-packages\glances\outputs\glances_curses.py", line 5
02, in display
    'processlist').get_stats_display(args=self.args)
  File "C:\Python27\lib\site-packages\glances\plugins\glances_plugin.py", line 5
51, in get_stats_display
    'msgdict': self.msg_curse(args),
  File "C:\Python27\lib\site-packages\glances\plugins\glances_processlist.py", l
ine 420, in msg_curse
    ret.extend(self.get_process_curses_data(p, first, args))
  File "C:\Python27\lib\site-packages\glances\plugins\glances_processlist.py", l
ine 220, in get_process_curses_data
    msg = ' {0:9}'.format(str(p['username'])[:9])
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 10:
 ordinal not in range(128)
```

I suppose this is due to the language of my OS ... are there any restrictions concerning language and are there any solution to this. I browsed the Glances doc and the Wiki pages with no success ... this is why I finally opened this issue.

**Comment by nicolargo** [comment]
Nope, it is clearly an issue.
String encoding is a mess in Python....

We will try to correct it as soon as possible.

Thanks for your bug report.

**Comment by nicolargo** [comment]
Can you try to edit the file C:\Python27\lib\site-packages\glances\plugins\glances_processlist.py and change the line 220 with:

    msg = ' {0:9}'.format(str(p['username'].encode('utf-8'))[:9])

**Comment by mohierf** [comment]
I will make your test ... and I confirm that on an English Windows workstation everything is fine!

I even tried on my French WS with an other account, because mine has é in the username ... but it does not run.

**Comment by nicolargo** [comment]
Ok thanks, wait your feedback.

**Comment by mohierf** [comment]
Here is it:

On the left, Glances on my Win8.1 French WS and on the right a Win7 English VM: 

![image](https://cloud.githubusercontent.com/assets/2932687/11009209/a2837352-84d5-11e5-8d61-c2739267d0c6.png)

**Comment by nicolargo** [comment]
Ok...
The Windows terminal is ugly...
Why don't you use the Web Interface (glances -w) ?

**Comment by mohierf** [comment]
I agree with this ... my intent is not to use the Windows terminal. I would like to use glances as a monitoring agent for Shinken.

**Comment by mohierf** [comment]
Bad luck with server mode: 
![image](https://cloud.githubusercontent.com/assets/2932687/11009355/868d6c1a-84d6-11e5-8b40-1a12c90bb9e4.png)

**Comment by nicolargo** [comment]
Mmmmhhh, never seen this kind of error on the Web Server mode...

**Comment by mohierf** [comment]
More info:

![image](https://cloud.githubusercontent.com/assets/2932687/11009486/87f468a0-84d7-11e5-987d-d0ae50eb89da.png)

Still some encoding problems ...

**Comment by nicolargo** [comment]
Ok... not easy to debug.
The encoding problem can be everywhere because the RESTFUL request is /all.
You have ran the test with the patched version (the one with the UTF-8 encoding for the username ?)

**Comment by mohierf** [comment]
Yes, with patched version ...

I am trying the server mode on Win7 US to compare but I have another problem:
![image](https://cloud.githubusercontent.com/assets/2932687/11009625/a6f8a832-84d8-11e5-98ee-fae42dddd476.png)


Did I missed some requirements ? I installed glances and bottle ...

**Comment by nicolargo** [comment]
Nope, you only need Glances, Psutil and Bottle.

I think that it is an issue on the new Web Interface. It should check that the 'nice' CPU property exist before trying to display it. @notFloran or @nclsHart : Can you have a look on this issue ? (perhaps it is better to create a new issue specificaly).

**Comment by mohierf** [comment]
Thanks for your help ... I will make more tests later

**Comment by nicolargo** [comment]
@mohierf : The fix #717 of the develop branch should solve the 'nice' CPU issue.

**Comment by asergi** [comment]
The problem here is the use of str() as a converter from unicode to bytes. In fact, if you try to just drop str() I guess it works. Anyway, if use develop apply this:
```diff
diff --git a/glances/plugins/glances_processlist.py b/glances/plugins/glances_processlist.py
index dd1d4a7..7f0f927 100644
--- a/glances/plugins/glances_processlist.py
+++ b/glances/plugins/glances_processlist.py
@@ -25,7 +25,7 @@ import os
 from datetime import timedelta
 
 # Import Glances libs
-from glances.core.compat import iteritems
+from glances.core.compat import b, iteritems
 from glances.core.glances_logging import logger
 from glances.plugins.glances_core import Plugin as CorePlugin
 from glances.core.glances_globals import is_windows
@@ -214,8 +214,8 @@ class Plugin(GlancesPlugin):
         # USER
         if 'username' in p:
-            # docker internal users are displayed as ints only, therefore str()
-            msg = ' {0:9}'.format(str(p['username'])[:9])
+            # docker internal users are displayed as ints only, therefore bytes
+            msg = ' {0:9}'.format(b(p['username'])[:9])
             ret.append(self.curse_add_line(msg))
         else:
             msg = ' {0:9}'.format('?')
```
